includes/class-wc-payments-account.php

```diff
@@ -570,6 +570,9 @@ public function maybe_redirect_to_server_link() {
 
 		try {
 			$link = $this->payments_api_client->get_link( $args );
+			if ( isset( $args['type'] ) && 'complete_kyc_link' === $args['type'] && isset( $link['state'] ) ) {
```

Comment:
it's needed similarly to https://github.com/Automattic/woocommerce-payments/blob/6f9dccab802da07e498c71bb059e18f8c04da5f6/includes/class-wc-payments-account.php#L1126-L1126. Without saving state in transient, we'll be redirected to payments/connect page because of https://github.com/Automattic/woocommerce-payments/blob/6f9dccab802da07e498c71bb059e18f8c04da5f6/includes/class-wc-payments-account.php#L1139-L1144